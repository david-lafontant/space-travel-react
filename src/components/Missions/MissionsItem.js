import React from 'react'

const MissionsItem = ({key,name,description}) => {
  return (
    <li>
        <h3>{name}</h3>
        <p>{description}</p>
    </li>
  )
}

export default MissionsItem
