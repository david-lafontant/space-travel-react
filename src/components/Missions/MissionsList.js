import React from 'react'
import { useSelector } from 'react-redux'
import MissionsItem from './MissionsItem'

const MissionsList = () => {
    const missions = useSelector((state)=> state.missions);
    console.log(missions)
  return (
    <ul>
        
        {missions.map((mission)=> (
           <MissionsItem
           name={mission.mission_name}
           key={mission.mission_id}
           description={mission.mission_description}
           />

        ))}
        </ul>
  )
}

export default MissionsList
