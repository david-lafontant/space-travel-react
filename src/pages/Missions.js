import React from 'react';
import MissionsList from '../components/Missions/MissionsList';

const Missions = () => (
  <div>
    <MissionsList />
  </div>
);

export default Missions;
