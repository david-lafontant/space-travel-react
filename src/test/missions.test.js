import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import elementStore from '../redux/configureStore';
import Missions from '../pages/Missions';

// Arangment
const missions = [
  {
    desc: 'The SpaceX Misions are best plan for system',
    key: 'cheers',
    id: 'csgo',
    type: 'hard',
    name: 'care',
    image: 'url',
    reservation: false,
  },
];

describe('Test render the Missions Component', () => {
  it('Missions section renders a hardcoded object, passed as a prop', () => {
    // Action
    const { container } = render(
      <Provider store={elementStore}>
        <Missions key={missions[0]} mission={missions[0]} />
      </Provider>,
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
