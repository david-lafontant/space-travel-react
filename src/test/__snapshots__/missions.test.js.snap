// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render the Missions Component Missions section renders a hardcoded object, passed as a prop 1`] = `
<div>
  <div
    style="width: 75%; margin-left: auto; margin-right: auto;"
  >
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Missions
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
